### ISSUE TITLE
Native crashes are not sent to On-Premise endpoint

### ISSUE THREAD

**Issue by troy-lamerton** [bugsnag/bugsnag-unity#195]
### Description

The custom endpoint should be used when a native crash is sent.

### Issue

After an app crash, the crash is not reported anywhere. I dont see it on bugsnag.com and I dont see it on the server.

### Environment

* bugsnag-unity version: v4.8.2
* Unity version: 2019.2.8f1
* Operating system name and version: Mac OS Catalina (10.15.4)
* Target platform names and versions: Android 10.0
* Initializing bugsnag via the Unity UI or in code? Code

### Example code snippet

```cs
Bugsnag.Init(demoApiKey, false);
// run a simple web server and put your local network ip here
Bugsnag.Configuration.Endpoint = new Uri("http://192.168.1.20:3000");
```

Exceptions reported with `Bugsnag.Notify()` are received on the web server. Native crashes are not. So I suspect this is an issue with the Endpoint not being passed to the Android plugin.

**Comment by troy-lamerton** [comment]
This is my hacky solution
https://github.com/thesingularitygroup/bugsnag-android/commit/d77890a1488a3901084d9b8f67b4245b53b17ee4

**Comment by mattdyoung** [comment]
Hi @troy-lamerton 

In your example code snippet you've initialized with:
```
Bugsnag.Init(demoApiKey, false);
```

The second parameter is the `autoNotify` setting and using `false` will disable all automatic reporting of unhandled errors and native crashes:
https://docs.bugsnag.com/platforms/unity/#reporting-unhandled-errors

Can you try with:
```
Bugsnag.Init(demoApiKey, true);
```
and confirm whether that behaves as expected?

**Comment by troy-lamerton** [comment]
I investigated this extensively and I can share what I discovered:

- `Bugsnag.Init(demoApiKey, false);`
native crashes are always reported, this is [documented here](https://docs.bugsnag.com/platforms/unity/configuration-options/#autonotify)
- changing the default endpoint in the android code [Configuration.java#L37](https://github.com/bugsnag/bugsnag-android/blob/69aba9a05e5de94101a03ebb24bf49d7d6438085/bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.java#L37) does affect java exceptions
- native crashes are not affected by changing the default endpoint in [Configuration.java#L37](https://github.com/bugsnag/bugsnag-android/blob/69aba9a05e5de94101a03ebb24bf49d7d6438085/bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.java#L37), somehow they still go to notify.bugsnag.com
- during this attempt, searching the entire bugsnag-android project for "notify.bugsnag.com" gave no results, but somehow the native crashes were still going there
- it is possible to override the endpoint just before delivery, in [DefaultDelivery.java#L57](https://github.com/bugsnag/bugsnag-android/blob/830f25a13221d196af58a8088f2c9e77e4a0cc13/bugsnag-android-core/src/main/java/com/bugsnag/android/DefaultDelivery.java#L57) and this works for native crashes

I'm not using the code anymore so I can't help with this any further.

**Comment by mattdyoung** [comment]
Hi @troy-lamerton 

Thanks for the additional detail. I've reproduced what you're seeing and it's definitely a bug so we'll aim to fix in an upcoming release.

Thanks again!

**Comment by yousif-bugsnag** [comment]
This is now fixed in [v5.0.0](https://github.com/bugsnag/bugsnag-unity/releases/tag/v5.0.0) 🎉
